### ISSUE TITLE
The image array the LoadImaged load NIfTI file not the same as I would get if I use SimpleITK library to load file and use GetArrayFromImage to get the image array

### ISSUE THREAD

**Issue by mchust** [Project-MONAI/MONAI#5798]
Hello everyone, I've just started to get acquainted with MONAILabel and am writing a liver segmentation app but using my pretrained model following the instructions in the README.md file located in the MONAILabel/sample-apps/radiology folder. After following the instructions I was able to load my pretrained model and run the inference with my dataset. However, I found that the inference results with the MONAILabel app are worse than the inference results if I run my Pytorch inference code. After spending 2 days debugging the code, I realized that the image array after using monai.transforms.LoadImaged to load my input (the NIfTi file) is not the same as the image array I would get if I used SimpleITK library to load the file and use the GetArrayFromImage function to get the image array. Upon further debugging, I found that using of np.ascontiguousarray in the convert_to_numpy function in the monai/utils/type_conversion.py folder changed the value of the array. Has anyone had the same problem as me and can help me solve this problem. Thank you very much!
![image](https://user-images.githubusercontent.com/115694996/210297340-1ffb0738-6637-4c77-b63f-6d22646ec1e2.png)
![image](https://user-images.githubusercontent.com/115694996/210297014-04187107-cbc7-4053-90eb-c45d37c5a542.png)

**Comment by Nic-Ma** [comment]
Hi @tangy5 ,

Could you please help take a look at this question?

Thanks in advance.

**Comment by tangy5** [comment]
Hi @mchust , I didn't directly tried, but the LoadImaged transform can use ITK reader, it's a wrapper class. 
For a quick solution, you can try `"ITKReader"` instead of "NibabelReader" for LoadImaged. 
I feel like the data type matters and incompatible with NibabelReader here. 

For details, see: 
https://docs.monai.io/en/stable/transforms.html#loadimaged

Hope this helps, thanks.

**Comment by mchust** [comment]
I have tried both "ITKReader" and "NibabelReader" reader types. However, the above problem will still appear. In addition, I noticed that if I use the convert_to_tensor function in the monai/utils/type_conversion.py file, the value of the image array will also change. The reason is Pytorch's torch.as_tensor function. Also the way I recognize the change in the value of the image array is to print the sum of the values of all the elements of the image array before and after going through the function np.ascontiguous or torch.as_tensor. Is my test correct? If yes, can you double check the convert_to_numpy and convert_to_tensor functions in the monai/utils/type_conversion.py file?

**Comment by tangy5** [comment]
Hi @Nic-Ma @wyli , 
This might be a general issue with `LoadImaged` transform.
Did we have this issue before? "The loaded image array using LoadImaged is different from plain SimpleITK"
convert_to_tensor function also has some issues per reported.

**Comment by mchust** [comment]
https://github.com/Project-MONAI/MONAI/blob/97918e46e0d2700c050e678d72e3edb35afbd737/monai/transforms/io/array.py#L275
I started debugging from here. After this line, the value of img_array seems to have changed. (The sum of all the elements of img_array before and after this line is not the same.)

**Comment by wyli** [comment]
maybe it's because of the memory layout and indexing convention differences,
```py
>>> import SimpleITK as sitk
>>> image = sitk.Image(256, 128, 64, sitk.sitkInt16)
>>> sitk.GetArrayFromImage(image).shape
(64, 128, 256)
```
the spatial dimensions are 'reversed' (https://github.com/Project-MONAI/MONAI/issues/2891, https://github.com/Project-MONAI/MONAI/issues/3021)

you can try `LoadImaged(keys="image", reader="itkreader", reverse_indexing=True)`
to keep the GetArrayFromImage behaviour.

if this doesnt address your issue, I'm happy to look into it further if you can upload an example image for debugging.

**Comment by mchust** [comment]
I have some updates for you. Where f1 is img_array before the above line of code and f2 is img_array after the above line of code.
https://github.com/Project-MONAI/MONAI/blob/97918e46e0d2700c050e678d72e3edb35afbd737/monai/transforms/io/array.py#L275
 I see that f1 and f2 have all the same values but I don't understand why when summing the parts from f1 and f2 are different. Can you help me explain?
This is my debug screen: 
![image](https://user-images.githubusercontent.com/115694996/210531616-7b061385-0fa5-4d52-b9f3-4299057c57fc.png)

**Comment by wyli** [comment]
what are the dtypes, f1.dtype and f2.dtype? because you have 1e8 elements, the cumulative error may become noticeable.

**Comment by mchust** [comment]
Thank you for your help. After adding reverse_indexing=True everything is fine. Thanks a lot !!!

**Comment by mchust** [comment]
I sincerely apologize for reopening this issue @wyli . With the addition of reverse_indexing=True, The img_array is correct only when I am debugging code. More detail, I use ipdb to debug code and I add 2 lines import ipdb; ipdb.set_trace() before and after this line of code 
https://github.com/Project-MONAI/MONAI/blob/d990ff5e6acab8655d507aeac934defd4f7a5fdc/monai/transforms/io/array.py#L282
![image](https://user-images.githubusercontent.com/115694996/211463468-562a2eea-af62-493f-bebb-c6e803b5da8b.png)
 However when I comment the above 2 lines of ipdb code to run the code, the img_array before entering infer is a zero array or if I try print(img) after the above line of code, my code was killed (Segmentation defaul()). 
![image](https://user-images.githubusercontent.com/115694996/211463653-5ba8976c-f8c5-4c0c-9d98-224fc4d3afd6.png)
![image](https://user-images.githubusercontent.com/115694996/211463798-1800db28-cd55-4753-9509-d60cd8bdbb97.png)
 Specifically if I set reverse_indexing=False this problem doesn't happen. I wonder if this is a problem with your code or with my hardware. I am using an A100 card (40GB) to run the code. Can you help me answer this question?

**Comment by wyli** [comment]
I couldn't reproduce this issue based on your comments... I'd recommend that you create a new bug report, following the template https://github.com/Project-MONAI/MONAI/issues/new?assignees=&labels=&template=bug_report.md&title= and making a small self-contained example to reproduce the problem. closing this ticket for now
